### ISSUE TITLE
[Demand] Image Encryption (Stage 2)

### ISSUE THREAD

**Issue by neruthes** [DimensionDev/Maskbook#1360]
## Metadata

Field | Value
--- | ---
Previous Stage | #1217 
Figma Link | https://www.figma.com/file/2ezfQbmv3uXMM4O7zSvZj1/Plugin-Image-Shuffle?node-id=1%3A91
Expected UX prototype due | 2020-W??-5
Actual UX prototype release | -
Expected engineering due | 2020-W??-5
Corresponding Pull Request | -
Completion Version | -

## Introduction

Thanks to @nazariyv, image shuffling has been made possible.

Now we will work on refactoring things to present this feature better.

## Changes on Plugin System Design

Documentation: https://confluence.dimension.chat/x/9IAf

A new section "Natural Post Integration" is added.

A Plugin may declare that it supports its Custom Post Type. This declaration is SNS-specific.

If the user has any such plugin, there will be a [dropdown menu](https://www.figma.com/file/4DmpRxZk4jDeqrfA5xrPCp/Dashboard-v3?node-id=2071%3A1370) next to the "Compose Encrypted Post" button, which includes the list of Plugins with such declaration.

A such Plugin can write arbitrary data to Natural Post composition view. A such Plugin can read Natural Posts. Maskbook identifies each Natural Post and attempts to decode data from it, and attempts to let every such Plugin interpret it.

Upon successful interpretation, a such Plugin renders a Snippet (Post-Affixing Canvas) in the Maskbook Interpreted Post view.

In short, Maskbook Core is no longer the only Natural Post Interpreter. If multiple Plugins report successful interpretations, multiple Snippets will be displayed under the Post.

## UX Prototype

https://www.figma.com/file/2ezfQbmv3uXMM4O7zSvZj1/Plugin-File-Service-Copy?node-id=1%3A91

## Notes

- None.

**Comment by Tedko** [comment]
After discussion w/ @OiCkilL the ` dropdown menu ` is not suitable for this `Image Shuffle`.  Have a complete discussion 1st @neruthes @OiCkilL

**Comment by neruthes** [comment]
Discussions are welcomed at https://github.com/DimensionDev/Maskbook-Talks/issues/50
